src/framework/filterControl/FilterItem.jsx

```diff
@@ -0,0 +1,158 @@
+
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import {
+  Filter,
+} from '../services';
+
+import Entity from '../services/string/Entity';
+import FilterForm from './forms/FilterForm.jsx';
+
+export default class FilterItem extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFormOpen: false,
+    };
+
+    this.onCancelEditing = this.onCancelEditing.bind(this);
+    this.onClickAnywhere = this.onClickAnywhere.bind(this);
+    this.onEditFilter = this.onEditFilter.bind(this);
+    this.onRemoveSelectedFilter = this.onRemoveSelectedFilter.bind(this);
+    this.onUpdateFilter = this.onUpdateFilter.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.onClickAnywhere);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.onClickAnywhere);
+  }
+
+  onCancelEditing() {
+    this.setState({
+      isFormOpen: false,
+    });
+  }
+
+  onClickAnywhere(event) {
+    if (
+      this.state.isFormOpen &&
+      !this.filterFormDropdown.contains(event.target) &&
+      !this.filterItemDiv.contains(event.target)
+    ) {
+      this.setState({
+        isFormOpen: false,
+      });
+    }
+  }
+
+  onEditFilter() {
+    this.setState({
+      isFormOpen: true,
+    });
+  }
+
+  onRemoveSelectedFilter() {
+    this.props.onRemoveSelectedFilter(this.props.filter);
+  }
+
+  onUpdateFilter(oldFilter, filter) {
+    this.props.onReplaceFilter(oldFilter, filter);
+    this.setState({
+      isFormOpen: false,
+    });
+  }
+
+  renderFilterForm() {
+    const onUpdateFilter =
+      filter => this.onUpdateFilter(this.props.filter, filter);
+
+    return (
+      <FilterForm
+        comparisonValue={this.props.filter.comparisonValue}
+        filterOption={this.props.filter.filterOption}
+        onAddFilter={onUpdateFilter}
+      />
+    );
+  }
+
+  renderForm() {
+    if (!this.state.isFormOpen) {
+      return undefined;
+    }
+
+    return (
+      <div
+        className="filterFormDropdown filterFormDropdown--editing"
+        ref={(div) => { this.filterFormDropdown = div; }}
+      >
+        <div className="filterFormDropdown__form">
+          <div className="filterFormDropdown__form__header">
+            <div className="filterFormDropdown__form__header__title">
+              {this.props.filter.filterOption.name}
+            </div>
+            <span
+              className="filterFormDropdown__form__header__closeButton"
+              onClick={this.onCancelEditing}
+            />
+          </div>
+          {this.renderFilterForm()}
+        </div>
+      </div>
+    );
+  }
+
+  renderFilterItem() {
+    const filterName = this.props.filter.filterOption.name;
+    const title =
+      `${filterName}: ${this.props.filter.humanizeComparisonValue()}`;
+
+    return (
+      <div
+        className="filterItem button"
```

Comment:
FYI: The button effect is only being used on actual buttons. Also the 'add' button doesn't have any effect. Still I like the idea of indicating that you can click, perhaps pointer is enough or the `box-shadow` effect on the other side with an ellipsis icon or so... just an idea, no need for change.